metis-enrichment/metis-enrichment-client/src/test/java/eu/europeana/enrichment/utils/EntityMergeEngineTest.java

```diff
@@ -0,0 +1,605 @@
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import java.io.UnsupportedEncodingException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import org.jibx.runtime.JiBXException;
+import org.junit.Test;
+import eu.europeana.corelib.definitions.jibx.AgentType;
+import eu.europeana.corelib.definitions.jibx.Alt;
+import eu.europeana.corelib.definitions.jibx.Concept.Choice;
+import eu.europeana.corelib.definitions.jibx.Lat;
+import eu.europeana.corelib.definitions.jibx.LiteralType;
+import eu.europeana.corelib.definitions.jibx.PlaceType;
+import eu.europeana.corelib.definitions.jibx.RDF;
+import eu.europeana.corelib.definitions.jibx.ResourceOrLiteralType;
+import eu.europeana.corelib.definitions.jibx.ResourceType;
+import eu.europeana.corelib.definitions.jibx.TimeSpanType;
+import eu.europeana.corelib.definitions.jibx._Long;
+import eu.europeana.enrichment.api.external.model.Agent;
+import eu.europeana.enrichment.api.external.model.Concept;
+import eu.europeana.enrichment.api.external.model.EnrichmentBase;
+import eu.europeana.enrichment.api.external.model.Label;
+import eu.europeana.enrichment.api.external.model.LabelResource;
+import eu.europeana.enrichment.api.external.model.Part;
+import eu.europeana.enrichment.api.external.model.Place;
+import eu.europeana.enrichment.api.external.model.Resource;
+import eu.europeana.enrichment.api.external.model.Timespan;
+
+public class EntityMergeEngineTest {
+
+  private static Place createPlace() {
+    Place place = new Place();
+
+    place.setAbout("aboutP1");
+    place.setAlt("123.456");
+
+    Label label1 = new Label("labelP1");
+    Label label2 = new Label("langP2", "labelP2");
+    place.setAltLabelList(Arrays.asList(new Label[] {label1, label2}));
+
+    Part part1 = new Part("partP1");
+    place.setHasPartsList(Arrays.asList(new Part[] {part1}));
+
+    Part part2 = new Part("partP2");
+    place.setIsPartOfList(Arrays.asList(new Part[] {part2}));
+
+    place.setLat("12.34");
+    place.setLon("43.21");
+
+    Label label3 = new Label("labelP3");
+    place.setNotes(Arrays.asList(label3));
+
+    Label label4 = new Label("langP4", "labelP4");
+    place.setPrefLabelList(Arrays.asList(new Label[] {label4}));
+
+    Part part3 = new Part("partP3");
+    Part part4 = new Part("partP4");
+    place.setSameAs(Arrays.asList(new Part[] {part3, part4}));
+
+    return place;
+  }
+
+  private static Place createFirstPlaceWithNullValues() {
+    Place place = new Place();
+    place.setAltLabelList(Arrays.asList(new Label[] {null}));
+    place.setHasPartsList(Arrays.asList(new Part[] {}));
+    place.setIsPartOfList(null);
+    place.setNotes(Arrays.asList(new Label()));
```

Comment:
Can be changed to Collections.singletonList(new Label()) since it's single.

Reply:
A singleton list is immutable/unmodifiable. I didn't want to presume that the functionality inside the objects can deal with an immutable list. I would suggest leaving this as-is.

Reply:
makes sense ok